### PR TITLE
Enhance Kafka integration tests to support multiple Kafka versions

### LINKED ISSUE #3068: Document and update integration test with the versions of Kafka Jaeger supports
## Requirement - what kind of business use case are you trying to solve?

Update [jaeger-v2 kafka integration tests][1]  to test with the two most recent versions of Kafka

## Problem - what in Jaeger blocks you from solving the requirement?

Right now we don't mention anywhere in the doc the version of kafka we support. This has lead to multiple open issues: https://github.com/jaegertracing/jaeger/issues/3059 https://github.com/jaegertracing/jaeger/issues/2950  https://github.com/jaegertracing/jaeger/issues/2068

Although [Sarama](https://github.com/Shopify/sarama/) says "Sarama is a Go library for Apache Kafka 0.8, and up." but their [functional tests](https://github.com/Shopify/sarama/blob/master/.github/workflows/ci.yml#L13) only test for kafka v2.7.1 and v2.8.0.

The dependabot or other PR bump up the version of Sarama which may break integration with older version of kafka.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Update kafka integration test for each kafka versions we support just like how we do for ES

## Any open questions to address

This will increase the integration test time but I think we should  do it to prevent unknown breaking changes.

[1]: https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/integration/kafka_test.go

### PR DESCRIPTION
**Pull request by zzzk1**
## Which problem is this PR solving?
- Resolves #3068
## Description of the changes
- Introduce old version kafka env
- Update e2e CI script
## How was this change tested?
- integration tests

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `npm run lint` and `npm run test`

### CODE REVIEW

**Comment by zzzk1** [comment]
I 'm not sure where to record the document.... 🤔

File: docker-compose/kafka-integration-test/v2/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+services:
```
Comment (yurishkuro) [inline docker-compose/kafka-integration-test/v2/docker-compose.yml:1]: let's get rid of `kafka-integration-test/` folder and keep everything under `kafka/`

File: .github/workflows/ci-e2e-kafka.yml [hunk 2]
```diff
@@ -17,8 +17,13 @@ jobs:
...
       fail-fast: false
       matrix:
-        jaeger-version: [v1, v2]  # Adjust if there are specific versions of Jaeger
-    name: kafka ${{ matrix.jaeger-version }}
+        jaeger-version: [v1, v2]
+        version:
+          - major: 3.x
+            distribution: kafka
+          - major: 2.x
+            distribution: kafka
```
Comment (yurishkuro) [inline .github/workflows/ci-e2e-kafka.yml:25]:
```suggestion
        kafka-version: ["3.x", "2.x"]
```
the "distribution" is unnecessary, the whole file is about kafka

File: .github/workflows/ci-e2e-kafka.yml [hunk 3]
```diff
@@ -31,12 +36,12 @@ jobs:
...
       id: test-execution
-      run: bash scripts/kafka-integration-test.sh -j ${{ matrix.jaeger-version }}
+      run: bash scripts/kafka-integration-test.sh -j ${{ matrix.jaeger-version }} -v ${{ matrix.version.major }}
 
     - name: Upload coverage to codecov
       uses: ./.github/actions/upload-codecov
       with:
         files: cover.out
-        flags: kafka-${{ matrix.jaeger-version }}
+        flags: ${{ matrix.version.distribution }}-${{ matrix.version.major }}-${{ matrix.version.jaeger }}
```
Comment (yurishkuro) [inline .github/workflows/ci-e2e-kafka.yml:47]: no such variable `matrix.version.jaeger`

File: docker-compose/kafka/docker-compose.yml [hunk 4]
```diff
@@ -6,7 +6,7 @@ services:
     environment:
       - ALLOW_ANONYMOUS_LOGIN=yes
   kafka:
-    image: index.docker.io/bitnami/kafka@3.7.0
+    image: bitnami/kafka:3.7.0
```
Comment (yurishkuro) [inline docker-compose/kafka/docker-compose.yml:9]: why does this file need to change?

File: scripts/kafka-integration-test.sh [hunk 5]
```diff
@@ -5,36 +5,49 @@
...
+
 parse_args() {
-  while getopts "j:Kh" opt; do
+  while getopts "j:v:Kh" opt; do
     case "${opt}" in
     j)
       jaeger_version=${OPTARG}
       ;;
+    v)
+      kafka_version=${OPTARG}
```
Comment (yurishkuro) [inline scripts/kafka-integration-test.sh:36]: check for valid values here, instead of parsing in L50

File: scripts/kafka-integration-test.sh [hunk 6]
```diff
@@ -5,36 +5,49 @@
 
 set -euf -o pipefail
 
-compose_file="docker-compose/kafka-integration-test/docker-compose.yml"
+compose_file="docker-compose/kafka-integration-test/v3/docker-compose.yml"
```
Comment (yurishkuro) [inline scripts/kafka-integration-test.sh:8]: this should be assigned later once kafka_version is parsed into valid values

**Comment by zzzk1** [comment]
@jpkrohling mention **nightly jobs** here: https://github.com/jaegertracing/jaeger/issues/3068#issuecomment-856572419, so what should we do now?😊

File: scripts/kafka-integration-test.sh [hunk 7]
```diff
@@ -5,36 +5,53 @@
...
 manage_kafka="true"
 success="false"
 
-print_help() {
-  echo "Usage: $0 [-K] -j <jaeger_version>"
+usage() {
+  echo "Usage: $0 [-K] -j <jaeger_version> -v <kafka_version>"
   echo "  -K: do not start or stop Kafka container (useful for local testing)"
   echo "  -j: major version of Jaeger to test (v1|v2)"
+  echo "  -v: kafka major version (3.9|2.8)"
```
Comment (yurishkuro) [inline scripts/kafka-integration-test.sh:18]: Specify the exact expected values. The exact version like 3.9 is controlled by the docker-compose file, not by this script.
```suggestion
  echo "  -v: kafka major version (3.x|2.x)"
```

File: scripts/kafka-integration-test.sh [hunk 8]
```diff
@@ -5,36 +5,53 @@
...
+
 parse_args() {
-  while getopts "j:Kh" opt; do
+  while getopts "j:v:Kh" opt; do
     case "${opt}" in
     j)
       jaeger_version=${OPTARG}
       ;;
+    v)
+      kafka_version=${OPTARG%%.*}
```
Comment (yurishkuro) [inline scripts/kafka-integration-test.sh:36]: I was thinking something like:
```
case ${OPTARG} in
3.x)
  kafka_version=v3
  ;;
2.x)
  kafka_version=v2
  ;;
*)
  echo "Error: Invalid Kafka version. Valid options are 3.x or 2.x"
  usage
  ;;
esac
```

**Review by yurishkuro — COMMENTED** [review]
please also update renovate.json

File: scripts/kafka-integration-test.sh [hunk 9]
```diff
@@ -5,38 +5,63 @@
...
       manage_kafka="false"
       ;;
     *)
-      print_help
+      usage
       ;;
     esac
   done
-  if [ "$jaeger_version" != "v1" ] && [ "$jaeger_version" != "v2" ]; then
+  if [[ "$jaeger_version" != "v1" && "$jaeger_version" != "v2" ]]; then
```
Comment (yurishkuro) [inline scripts/kafka-integration-test.sh:57]: don't have objection to the change, but can you elaborate why you wanted to change it?

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**